src/Components/Classes/Form/index.jsx

```diff
@@ -55,20 +90,29 @@ const Form = ({ onSubmit, onCancel, editMode, classe }) => {
       activity: activity,
       slots: parseInt(slots)
     };
-    onSubmit(formData);
+    createSubmit(formData);
     setDay([]);
     setHour('');
     setTrainer('');
     setActivity('');
     setSlots('');
+    history.push('/classes');
```

Comment:
This line should be in the success block of each response (PUT, POST). In case of error, you should not exit the form.